File: split/newkern.pas
function newkern(w: scaled): halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 11;
  mem[p].hh.b1 := 0;
  mem[p + 1].int := w;
  newkern := p;
end;

File: cpp/newkern.cpp
#include "newkern.h"
#include "getnode.h"

halfword newkern(scaled w)
{
	auto p = getnode(2);
	type(p) = kern_node;
	subtype(p) = 0;
	width(p) = w;
	return p;
}
